### PR TITLE
[FLINK-26794][tests] Use API to access job checkpoints in IT cases

### PR DESCRIPTION
**Pull request by rkhachatryan**
```
Using dedicated API has the following advantages over looking for checkpoints on FS:
- less prone to NoSuchFileException
- less brittle because file layout is not a declared API
- less IO usage
```

### CODE REVIEW

**Comment by rkhachatryan** [comment]
@AHeise, @fapaul I think we discussed and/or fixed `FileNotFoundException in multiple IT cases some time before.
WDYT about this approach, e.g. using API instead of traversing the FS?

**Comment by rkhachatryan** [comment]
Test failure unrelated (FLINK-26882).

File: flink-runtime/src/test/java/org/apache/flink/runtime/testutils/CommonTestUtils.java [hunk 1]
```diff
@@ -359,6 +364,53 @@ public static void waitForSubtasksToFinish(
...
+     */
+    public static String getLatestCompletedCheckpointPath(JobID jobID, MiniCluster cluster)
+            throws FlinkJobNotFoundException, NoSuchElementException, ExecutionException,
+                    InterruptedException {
+        return getLatestCompletedCheckpointPathMaybe(jobID, cluster)
+                .<NoSuchElementException>orElseThrow(
+                        () -> {
+                            throw new NoSuchElementException("No checkpoint was created yet");
+                        });
+    }
```
Comment (AHeise) [inline flink-runtime/src/test/java/org/apache/flink/runtime/testutils/CommonTestUtils.java:395]: I'd remove this method, the Optional one is easy enough to use, which I'd rename to `getLatestCompletedCheckpointPath`.

**Review by AHeise — APPROVED** [review]
LGTM. Thanks for the clean approach. I left a small comment that is probably a matter of taste.

**Comment by rkhachatryan** [comment]
Thanks for the review, I've inlined the method as you suggested.
Will merge the PR once the CI tests pass.

**Pull request merged**